### ISSUE TITLE
Add optional parameter names to various Pints plots

### ISSUE THREAD

**Issue by ben18785** [pints-team/pints#1072]
This would make some of the plots:

- traceplots
- pairwise plots

More immediately presentable / publishable...

**Comment by chonlei** [comment]
This was somewhat raised previously via adding it to the model #816, where things can be automated a bit more(?) and for saved samples CSV files as well? But yeah might be (a lot) easier to just do it via the plots as an optional input...

**Comment by DavAug** [comment]
I am using the existing notebooks to test whether naming the parameters works as expected. Would you like me to include the changes to those notebooks into this issue as well?

**Comment by ben18785** [comment]
Cool, thanks. No need to do it in all cases, but for a few notebooks would
be great.

**Comment by DavAug** [comment]
PR #1118

**Comment by DavAug** [comment]
I get alert that `assertRaiseRegexp` is going to be deprecated soon, and `assertRaiseRegex` should be used instead. I kept it for now to be in accordance with the rest of the tests.

Would you like me to replace it everywhere?

**Comment by DavAug** [comment]
I also changed the `exclude: env` by `exclude: venv` in the .flake8 file, as .gitignore seems to ignore a virtual environment named venv and not env, but I can revert that if that was intentional.

**Comment by MichaelClerx** [comment]
We usually have a block like this near the top

```
# Unit testing in Python 2 and 3
try:
    unittest.TestCase.assertRaisesRegex
except AttributeError:
    unittest.TestCase.assertRaisesRegex = unittest.TestCase.assertRaisesRegexp
```
and then use `assertRaisesRegex` exclusively

**Comment by chonlei** [comment]
Actually, thought we were now going for a customising by user like: https://github.com/pints-team/pints/blob/master/examples/plotting/customise-pints-plots.ipynb instead?

**Comment by MichaelClerx** [comment]
Both are valuable!

**Comment by MichaelClerx** [comment]
Closed in #1126 #1118

**Comment by MichaelClerx** [comment]
Thanks @DavAug !
